citest/test.js

```diff
@@ -322,3 +322,25 @@ describe('Event tests', function() {
     });
   });
 });
+
+describe('Write error tests', function() {
+  it('calls callback with error set when stream is not writeable', function(done) {
+    u.timeoutFilter(this, 10);
+    const dirname = fs.mkdtempSync('test_backup');
+    // make temp dir not writeable
+    fs.chmodSync(dirname, 0);
+    const filename = dirname + '/test.backup';
+    const backupStream = fs.createWriteStream(filename, {flags: 'w'});
+    const params = {useApi: true};
+    // try to do backup and check err was set in callback
+    u.testBackup(params, 'animaldb', backupStream, function(err) {
+      // error should have been set
+      assert.ok(err != null);
+      assert.equal(err.code, 'EACCES');
+      // cleanup temp dir
```

Comment:
The asserts will throw if they fail which will cause an `uncaught` exception in the tests.
This style worked more cleanly for me and made sure that the after hooks etc were not interrupted.
```js
try {
  assert ...
  done()
} catch (err) {
  done(err);
}
```
You will also probably want to move the cleanup before the asserts, otherwise it won't run in cases where the test fails.